src/librustc/mir/interpret/value.rs

```diff
@@ -138,14 +146,53 @@ impl<'tcx> Scalar {
         }
     }
 
+    #[inline]
+    pub fn is_null(self) -> bool {
+        match self {
+            Scalar::Bits { bits, .. } => bits == 0,
+            Scalar::Ptr(_) => false
+        }
+    }
```

Comment:
Uh... when did this happen? I feel it wasn't present in my original review, but I may have missed it.

I don't like this function because for pointers, it cannot reliably tell you whether this is NULL or not. There might have been overflow. That's why I deliberately *removed* this function recently.

Reply:
Oh lol, never mind, I am looking at the wrong PR. We have too many PRs called "miri refactor".^^